test/cases/channel/close.test.ts

```diff
@@ -0,0 +1,147 @@
+import setupTestEnvironment from '../../../src';
+import mediated from '../../../topologies/mediated.json';
+import { tokenAddresses, toWei } from '../../../src/util/token';
+import { LuminoTestEnvironment } from '../../../src/types/lumino-test-environment';
+import { Dictionary } from '../../../src/util/collection';
+import { LuminoNode } from '../../../src/types/node';
+import { verifyChannel } from '../../utils';
+import {
+    SETUP_TIMEOUT,
+    TEARDOWN_TIMEOUT,
+    ChannelState,
+    State,
+    TEST_TIMEOUT,
+} from '../../common';
+
+interface closeParams {
+    token: string;
+    partner: string;
+}
+
+describe('channel close', () => {
+    let nodes: Dictionary<LuminoNode>;
+    let env: LuminoTestEnvironment;
+
+    beforeAll(async () => {
+        env = await setupTestEnvironment(mediated);
+        nodes = env.nodes as Dictionary<LuminoNode>;
+    }, SETUP_TIMEOUT);
+
+    afterAll(async () => {
+        await env.stop();
+    }, TEARDOWN_TIMEOUT);
+
+    it(
+        'from initiator node',
+        async () => {
+            // deposit for node "initiator"
+            const initiatorDeposit = toWei(
+                mediated.channels[0].participant1.deposit // should be inferred
```

Comment:
these should eventually be fixed under [RLp-1084](https://jirainfuy.atlassian.net/browse/RLP-1084)

Reply:
>     1. About the numeric index it's because the defined topology, actually an alternative to that doesn't exists since in the simple topology we define general properties and a total number of nodes. Since those nodes are not named you must to call them by index, that's why. The responsible for the tests is who should check if the topology is simple or advanced, so i disagree in that point. We should anyway to create a validator or something to check if those indexes are accessible or not before running any test.

i agree, specially on your last point. i am fine with leaving this as-is until we iterate on the topology (or an eventual validator for it).

>     2. I'm not aware about the "performance limitations" that @mauricioirace is talking about but we shouldn't be storing any state in the sdk or either any of the wrapper objects, actually if we want to check the balance we should always call the node with the sdk and check the actual balance of the channel in that moment, those calls are not heavy at all, actually the channel call is really lightweight.

if we have no other choice but to call the SDK for obtaining some data, then that's that.

but, leaving performance aside: if through the specified topology we already know what the supposed balance (or deposit, etc.) should be, then why would we call the SDK to obtain this information?

i think that specially in tests, we should always trust our own test case data over what the SDK might say, and this is what i tried to convey in my last message: hard-coded, expected values make the tests more robust in comparison to calculated ones, moreso if they are calculated by the component we are testing.

so in summary, i have this question: why should we use the SDK to obtain a value we can deduct from the test-case data, such as the topology?